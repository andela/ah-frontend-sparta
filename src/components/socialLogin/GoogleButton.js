/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React, { Component } from 'react';
import GoogleLogin from 'react-google-login';
import FacebookLogin from 'react-facebook-login';
import { ClientId, google_social_url } from './GoogleConfig';

class Google extends Component {

  render() {
    const responseGoogle = (response) => {
        console.log(response);
        this.setState({ googleToken: response })
        const accessToken = response['tokenId'];
        console.log(accessToken);
        fetch(google_social_url, {
            headers: { "Content-Type": "application/json" },
            method: 'POST',
            body: JSON.stringify({user_token:{
                auth_token:accessToken
            }})

        }).then(result => result.json()).then(resp=>{console.log(resp)})
    }
    const responseFacebook = (response) => {
        console.log(response);
    }

    return (
      <div>
        <h2>LOGIN WITH FACEBOOK AND GOOGLE</h2>

        <FacebookLogin
        appId="" //APP ID NOT CREATED YET
        fields="name,email,picture"
        callback={responseFacebook}
        />

        <GoogleLogin
        clientId= {ClientId} 
        buttonText="LOGIN WITH GOOGLE"
        onSuccess={responseGoogle}
        onFailure={responseGoogle}
      />
      
      </div>
    )
  }
}

export default Google;
