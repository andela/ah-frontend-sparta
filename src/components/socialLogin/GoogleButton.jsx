import React from 'react';
import PropTypes from 'prop-types';
import GoogleLogin from 'react-google-login';
import { ClientId } from './GoogleConfig';
import './Google.css';

const Google = ({ responseGoogle }) => (
  <>
    <div className="google-login">
      <GoogleLogin
        clientId={ClientId}
        render={renderProps => (
          <span
            onClick={renderProps.onClick}
            disabled={renderProps.disabled}
            className="fab fa-google-plus"
            role="presentation"
          />
        )}
        redirectUri={process.env.redirectUri}
        onSuccess={responseGoogle}
        onFailure={responseGoogle}
      />
    </div>
  </>
);

Google.propTypes = {
  responseGoogle: PropTypes.func,
};

Google.defaultProps = {
  responseGoogle: () => {},
};

export default Google;
