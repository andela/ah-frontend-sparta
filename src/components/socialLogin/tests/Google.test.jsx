import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GoogleButton from '../GoogleButton';
import SignUpPage from '../../../pages/Signup/index';

describe('Google sign in', () => {
  it('should render without crashing', () => {
    const responseGoogle = {
      user_token: {
        auth_token: 'test_token',
      },
    };
    const wrapper = renderer.create(
      <GoogleButton responseGoogle={responseGoogle} />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe('Signup', () => {
  it('should should render correctly', () => {
    const wrapper = shallow(<SignUpPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
