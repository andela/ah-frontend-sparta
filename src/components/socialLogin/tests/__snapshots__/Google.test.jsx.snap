// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google sign in should render without crashing 1`] = `
<div
  className="google-login"
>
  <span
    className="fab fa-google-plus"
    disabled={true}
    onClick={[Function]}
    role="presentation"
  />
</div>
`;

exports[`Signup should should render correctly 1`] = `ShallowWrapper {}`;
